### ISSUE TITLE
Custom standard schemes should be registered as secure schemes

### ISSUE THREAD

**Issue by pfrazee** [electron/electron#7670]
- Electron version: 1.4.3
- Operating system: MacOS 10.12

I created an app demonstrating the issue: https://github.com/pfrazee/electron-bug-securewebviewcrash. Let me know if you have trouble reproducing.

This occurs around a webview:

``` html
<webview id="theWebview" src="custom://webview.html" preload="./webview-preload.js"></webview>
```

The webview opens a custom protocol, which is registered as follows:

``` js
protocol.registerStandardSchemes(['custom'])
app.on('ready', () => {
  protocol.registerHttpProtocol('custom', (req, cb) => {
    cb({ method: 'GET', url: 'http://localhost:12345' })
  }, () => console.log('Registered "custom" protocol'))
})
http.createServer((req, res) => res.end('Hello, world. This is the webview')).listen(12345)
```

The webview has a preload script which registers it as a secure protocol:

``` js
var { webFrame } = require('electron')
webFrame.registerURLSchemeAsPrivileged('custom')
```

For some reason, this combination causes devtools to crash the webview when the Application tab is opened. I used the more fine-grained controls in https://github.com/electron/electron/pull/7665 to isolate this specifically to the "secure" flag.

**Comment by pfrazee** [comment]
I finally have a custom build of Electron setup, so I'd be happy to help debug this, but I'll need some pointers. The webview 'crashed' event doesn't give any detail about what went wrong. Is there a way to get a stacktrace?

**Comment by deepak1556** [comment]
Currently access to cache storage apis will check if the origin is secure on both renderer and browser side. It will pass check on renderer since we call `registerURLSchemeAsSecure` but fails on browser and terminates renderer with bad access. Embedders have [ContentClient::AddSecureSchemesAndOrigins](https://cs.chromium.org/chromium/src/content/public/common/content_client.h?l=153) to fix this.

Non-standard custom schemes will still suffer from this crash since the check involves uniqueness of the [origin](https://cs.chromium.org/chromium/src/content/browser/cache_storage/cache_storage_dispatcher_host.cc?l=63). Will see if we can disable cache storage.

@pfrazee if you are planning to fix this, can override above mentioned method and insert schemes when `protocol.registerStandardSchemes` is called. Can follow how service worker schemes are registererd.

**Comment by pfrazee** [comment]
@deepak1556 Thanks, I'll see what I can do.

**Comment by kevinsawicki** [comment]
Fixed by #7947

**Comment by kevinsawicki** [comment]
@pfrazee so is this resolved by #7947 ?

**Comment by pfrazee** [comment]
@kevinsawicki yep!
